src/btree/row_srch.c

```diff
@@ -218,10 +218,10 @@ restart:	page = current->page;
 		}
 
 		/*
-		 * Binary search of the internal page.  There are two versions
-		 * (a default loop and an application-specified collation loop),
-		 * because moving the collation test and error handling inside
-		 * the loop costs about 5%.
+		 * Binary search of the internal page. Three versions (record
+		 * numbers, no application-specified collation order and
+		 * application-specified collation order), because moving those
+		 * tests and the error handling inside the loop costs about 5%.
```

Comment:
I didn't catch this comment change in my branch, sorry -- it should be updated to say something like "short keys with no application-specified collation order" instead of "record numbers".